src/Hangman/RandomWord.hs

```diff
@@ -24,4 +26,13 @@ randomWord wl = do
 
 gameWords :: WordList -> WordList
 gameWords wl =
-    filter (\w -> elem (length w) [5..9]) wl
+    foldr filter wl gameFilters
```

Comment:
Hmm not quite. I might've written it this way:

```hs
filter (\w -> all ($ w) gameFilters) wl
```

But your fold is better I think.